### PR TITLE
feat(misc): optionally update package.json scripts in init generators

### PR DESCRIPTION
**Pull request by leosvelperez**
## Related Issue(s)

Fixes #

### CODE REVIEW

**Review by isaacplmann — APPROVED** [review]

**Review by mandarini — APPROVED** [review]

File: packages/devkit/src/utils/update-package-scripts.ts [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+  );
+
+  const targetCommands = getInferredTargetCommands(result);
+  if (!targetCommands.length) {
+    return;
+  }
+
+  // exclude package.json scripts from nx
+  packageJson.nx ??= {};
+  packageJson.nx.includedScripts ??= [];
```
Comment (leosvelperez) [inline packages/devkit/src/utils/update-package-scripts.ts:65]: @FrozenPandaz @jaysoo do we want to include all scripts by default and only remove the ones we end up replacing?

Right now, the PR excludes everything if there is no `nx` entry in the `package.json`. If there's an `nx` entry, it's respected, but replaced scripts are removed.

File: packages/nx/src/command-line/add/add.ts [hunk 2]
```diff
@@ -104,7 +107,18 @@ async function initializePlugin(pkgName: string): Promise<void> {
...
 
   try {
-    await runNxAsync(`g ${pkgName}:${initGenerator}`);
+    let updatePackageScripts: boolean;
+    if (options.updatePackageScripts !== undefined) {
+      updatePackageScripts = options.updatePackageScripts;
+    } else {
+      updatePackageScripts =
+        process.env.NX_PCV3 === 'true' && coreNxPlugins.includes(pkgName);
+    }
```
Comment (leosvelperez) [inline packages/nx/src/command-line/add/add.ts:116]: @jaysoo @FrozenPandaz should we ignore `--updatePackageScripts` if no `NX_PCV3=true`?

Comment (jaysoo) [reply packages/nx/src/command-line/add/add.ts:116]: Let's keep it as is, we will be removing the flag later anyway.

**Review by jaysoo — APPROVED** [review]

**Review by FrozenPandaz — APPROVED** [review]
This is a good start. Let's iterate off of this.

**Pull request merged**